ansible/roles/zookeeper/tasks/start-zookeeper.yml

```diff
@@ -19,7 +19,13 @@
   file: path={{ default_data_dirs[0] }}/zookeeper state=directory
 - name: "ensure myid file is set"
   template: src=roles/zookeeper/templates/myid dest={{ default_data_dirs[0] }}/zookeeper/myid
+- name: Check if Zookeeper is running
```

Comment:
I don't recall any issues related to idempotency of the subsequent "start zookeeper" task, so I'm curious why we need to add this. Could you clarify?

Reply:
Sure. I have added the check because in ZK 3.5.6 the start script throws a non-zero return code when ZK was found already running which fails the task. It is slightly different in ZK 3.4.14 where the script was returning "0" return code in that situation. This check works for both versions of ZK & also the approach in ZK 3.5.6 is in-line with other start scripts in Hadoop.